lib/third_party/tinycbor/compilersupport_p.h

```diff
@@ -77,14 +80,18 @@
 #endif
 #define DBL_DECIMAL_DIG_STR     STRINGIFY(DBL_DECIMAL_DIG)
 
+#if defined(__GNUC__) && defined(__i386__) && !defined(__iamcu__)
+#  define CBOR_INTERNAL_API_CC          __attribute__((regparm(3)))
+#elif defined(_MSC_VER) && defined(_M_IX86)
+#  define CBOR_INTERNAL_API_CC          __fastcall
+#else
+#  define CBOR_INTERNAL_API_CC
+#endif
+
 #ifndef __has_builtin
 #  define __has_builtin(x)  0
 #endif
 
-
-/* Disable this optimization for TI ARM compiler v18 or higher because it has some issues
-with these intrinsics. */
-#if !defined(__TI_COMPILER_VERSION__) || __TI_COMPILER_VERSION__ < 18000000
```

Comment:
Deleting this intrinsic might cause issues on the TI platform so we should definitely run tests on TI (and all other platforms).

Reply:
Good catch. I blindly deleted all the old files. Let me try to run OTA E2E test on TI and see what happens